pol-core/bscript/compiler/Report.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Report.h"
+
+#include "../clib/logfacility.h"
+
+#include "compiler/file/SourceLocation.h"
+
+namespace Pol
+{
+namespace Bscript
+{
+namespace Compiler
+{
+Report::Report( bool display_warnings )
+  : display_warnings( display_warnings ), errors( 0 ), warnings( 0 )
+{
+}
+
+void Report::report_error( const SourceLocation& source_location, const char* msg )
+{
+  ++errors;
+  ERROR_PRINT << source_location << ": error: " << msg;
+}
+
+void Report::report_warning( const SourceLocation& source_location, const char* msg )
+{
+  ++warnings;
+  ERROR_PRINT << source_location << ": warning: " << msg;
+}
+
```

Comment:
Is "msg" always ending in a newline or you want to enforce that here?

Reply:
And for a second I thought `display_warnings` was being ignored, but saw the `error()` and `warning()` methods in the include. ;)

Reply:
Why not use std::string as type of msg, const char* always leads to ugly c_str() calls

Reply:
Yeah, `msg` is intended to always end in a newline.  This follows the pattern. of `compiler_error` and `compiler_warning`, most call site of which end in a newline.

Reply:
The `const char*` passed to these private methods always comes from a `fmt::Writer`, for example Report::error or Report::warning.  What's the preferred way of writing the buffer from a `fmt::Writer` to ERROR_PRINT or INFO_PRINT?